src/bin/trig.rs

```diff
@@ -316,80 +272,158 @@ impl SynthLanguage for Math {
+                egraph.add_expr(&"(sin 0)".parse().unwrap());
+                egraph.add_expr(&"(sin (/ PI 6))".parse().unwrap());
+                egraph.add_expr(&"(sin (/ PI 4))".parse().unwrap());
+                egraph.add_expr(&"(sin (/ PI 3))".parse().unwrap());
+                egraph.add_expr(&"(sin (/ PI 2))".parse().unwrap());
+                egraph.add_expr(&"(sin PI)".parse().unwrap());
+                egraph.add_expr(&"(sin (* 2 PI))".parse().unwrap());
+            }
 
-        // i * i = -1;
-        let i_id = egraph.add(Math::ComplexConst(complex_const_symbol("i")));
-        let i2_id = egraph.add(Math::CMul([i_id, i_id]));
-        let neg1_id = egraph.add(Math::ComplexConst(Complex::from("-1C")));
-        egraph.union(i2_id, neg1_id);
+            if allowedp("cos") {
+                egraph.add_expr(&"(cos 0)".parse().unwrap());
+                egraph.add_expr(&"(cos (/ PI 6))".parse().unwrap());
+                egraph.add_expr(&"(cos (/ PI 4))".parse().unwrap());
+                egraph.add_expr(&"(cos (/ PI 3))".parse().unwrap());
+                egraph.add_expr(&"(cos (/ PI 2))".parse().unwrap());
+                egraph.add_expr(&"(cos PI)".parse().unwrap());
+                egraph.add_expr(&"(cos (* 2 PI))".parse().unwrap());
+            }
 
-        // 2i * 2i = -4;
-        let twoi_id = egraph.add(Math::ComplexConst(complex_const_symbol("2i")));
-        let twoi2_id = egraph.add(Math::CMul([twoi_id, twoi_id]));
-        let neg4_id = egraph.add(Math::ComplexConst(Complex::from("-4C")));
-        egraph.union(twoi2_id, neg4_id);
+            if allowedp("tan") {
+                egraph.add_expr(&"(tan 0)".parse().unwrap());
+                egraph.add_expr(&"(tan (/ PI 6))".parse().unwrap());
+                egraph.add_expr(&"(tan (/ PI 4))".parse().unwrap());
+                egraph.add_expr(&"(tan (/ PI 3))".parse().unwrap());
+                // egraph.add_expr(&"(tan (/ PI 2))".parse().unwrap());
+                egraph.add_expr(&"(tan PI)".parse().unwrap());
+                egraph.add_expr(&"(tan (* 2 PI))".parse().unwrap());
+            }
+        }
 
-        // rewrites
+        // rewrites and extra rewrites
         synth.lifting_rewrites = vec![
-            rewrite!("def-cos"; "(cos ?a)" <=> "(Re (/C (+C (cis ?a) (cis (~ ?a))) 2C))"),
-            rewrite!("def-sin"; "(sin ?a)" <=> "(Re (/C (-C (cis ?a) (cis (~ ?a))) 2iC))"),
-            rewrite!("def-cis-add"; "(cis (+ ?a ?b))" <=> "(*C (cis ?a) (cis ?b))"),
+            // definition of sine, cosine, tangent
+            rewrite!("def-sin"; "(sin ?a)" <=> "(/ (- (cis ?a) (cis (~ ?a))) (* 2 I))"),
+            rewrite!("def-cos"; "(cos ?a)" <=> "(/ (+ (cis ?a) (cis (~ ?a))) 2)"),
+            rewrite!("def-tan"; "(tan ?a)" <=> "(* I (/ (- (cis (~ ?a)) (cis ?a)) (+ (cis (~ ?a)) (cis ?a))))"),
+
+            // definition of cosecant, secant, cotangent
+            rewrite!("def-csc"; "(csc ?a)" <=> "(/ 1 (sin ?a))"),
```

Comment:
why do we not define them in terms of `cis` etc?

Reply:
these seem like something we should expect ruler to actually learn, given the denotation using `cis, I, etc.`

Reply:
See note above